packages/core/src/axis-chart.ts

```diff
@@ -60,23 +60,27 @@ export class AxisChart extends Chart {
 			}
 		};
 
+		const isEnabled = this.model.getOptions().legend.enabled !== false;
+
 		// Decide the position of the legend in reference to the chart
 		let fullFrameComponentDirection = LayoutDirection.COLUMN;
-		const legendPosition = Tools.getProperty(this.model.getOptions(), "legend", "position");
-		if (legendPosition === LegendPositions.LEFT) {
-			fullFrameComponentDirection = LayoutDirection.ROW;
+		if (isEnabled) {
```

Comment:
So pending some additions I need to make to the `Chart` class (being able to modify the list of components in every render), we'd need to move the `isEnabled` logic into the actual legend component for now.

The reason being that in this case if you do enable the legend after having disabled it, it won't take effect if the logic is inside this file.

So the way it would work for now is the legend component would always get included, and the data-binding should get rid of all the rendered elements inside once we do disable.